### PR TITLE
ESFormats additions and fixes

### PR DESCRIPTION
**Pull request by leoetlino**
* Add a SignedBlobReader (base reader for signed blobs, like TMDs, tickets and certs) and move some of the common code into this base class.
* Add a CertReader and a cert chain parser that will be used by ES to verify containers.

### CODE REVIEW

**Comment by goldenercobra** [comment]
Related idea but maybe not pertaining to the scope of this PR; use the new [std::byte](http://en.cppreference.com/w/cpp/types/byte) for `m_bytes`?

File: Source/Core/Core/IOS/ES/Formats.cpp [hunk 1]
```diff
@@ -563,5 +614,89 @@ u32 UIDSys::GetOrInsertUIDForTitle(const u64 title_id)
...
+
+PublicKeyType CertReader::GetPublicKeyType() const
+{
+  const size_t offset = GetSignatureSize() + offsetof(CertHeader, public_key_type);
+  return static_cast<PublicKeyType>(Common::swap32(m_bytes.data() + offset));
+}
+
+std::vector<u8> CertReader::GetPublicKey() const
+{
+  static const std::map<SignatureType, std::pair<size_t, size_t>> type_to_key_info = {{
```
Comment (lioncash) [inline Source/Core/Core/IOS/ES/Formats.cpp]: `<map>` needs to be included.

**Comment by leoetlino** [comment]
@goldenercobra That'd be really outside of the scope of this PR, which is mostly code reorganisation + a new reader.

File: Source/Core/Core/IOS/ES/Formats.cpp (hunk 1 above)
Comment (leoetlino) [inline Source/Core/Core/IOS/ES/Formats.cpp]: Done

**Pull request merged**